appinventor/codelabs/mole-mash/index.html

```diff
@@ -0,0 +1,242 @@
+    }
+  </style>
+
+</head>
+<body unresolved class="fullbleed">
+
+  <google-codelab title="Mole Mash Codelab" feedback-link="https://goo.gl/forms/9zDi1V3arg167lIY2">
+
+      <google-codelab-step label="Overview">
+        <p>In this codelab, you&#39;ll create a game similar to Whack-a-Mole!
+        <h3 class="checklist">What you&#39;ll learn</h3>
+        <ul class="checklist">
+          <li>Use new components such as <b>image sprites</b> and the <b>Clock</b> component.</li>
+          <li>Use <b>procedures</b> to simplify your code.</li>
+          <li>Use new blocks such as <b>random</b> and <b>text</b> blocks.</li>
+        </ul>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Create a new project">
+        <p>From the <b>Projects</b> drop down menu, click <b>Start a new project</b>.
+        <p>Create a <a href="http://ai2.appinventor.mit.edu/" target="_blank">new project</a> called
+          <code>MoleMash</code> (no spaces).</p>
+        <img src="img/new_project.png">
+        <p>Set the screen's <b>Title</b> to "MoleMash".</p>
+        <p>Download <a href="http://appinventor.mit.edu/explore/sites/all/files/ai2tutorials/moleMash/mole.png" target="_blank">this picture of a mole</a> and upload it in the <b>Media</b> panel.</p>
+        <img src="img/mole_pic.png" style="max-width: 400px">
+
+      </google-codelab-step>
+
+      <google-codelab-step label="The first components">
+        <p>You should be familiar with adding various components in the <b>Designer View</b>. Add the following to your app:</p>
+          <ul>
+            <li>A <b>Canvas</b> named "MyCanvas". This is the area where the mole moves. Set its dimensions to 300 pixels wide by 300 pixels high.</li>
+            <li>A <b>Label</b> named "ScoreLabel" that shows the score, i.e., the number of times the player has hit the mole. Set the Text to "Score: ---".</li>
+            <li>A <b>Button</b> named "ResetButton".</li>
+            <li>A <b>Sound</b> component and name it "Noise". You'll use Noise to make the phone vibrate when the mole is hit.</li>
+          </ul>
+        <p>Your app should look something like this:</p>
+        <img src="img/ui1.png">
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add a timer">
+        <p>In order to make the mole jump periodically, you'll need a timer. Add a <b>Clock</b> component and name it "MoleTimer"; this component will go into the <b>non-visible components</b> area. Set the <b>TimeInterval</b> property to 500 milliseconds, and make sure that <b>TimerEnabled</b> is checked.</p>
+        <img src="img/timer.png">
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add a mole">
+        <p>To add the moving mole, we'll use a <b>sprite</b>. Sprites are images that can move on the screen within a Canvas. For MoleMash, our sprite won't move by itself; instead, you'll be setting the mole's position each time the timer fires.</p>
```

Comment:
~our sprite won't move by itself;~

This might be confusing to the students. Best to leave it out.